### PR TITLE
Core: Preserve kind load order on HMR when no sortFn is provided

### PR DESCRIPTION
**Pull request by shilman**
Issue: #9419 

## What I did

Preserve kind order when stories are edited.

## How to test

See attached test case. Also edit stories that are not last in the list in a project where no sort order is specified.

### CODE REVIEW

File: lib/client-api/src/story_store.ts [hunk 1]
```diff
@@ -131,6 +131,17 @@ export default class StoryStore extends EventEmitter {
       if (index && this._data[index].parameters.options.storySort) {
         const sortFn = this._data[index].parameters.options.storySort;
         stable.inplace(stories, sortFn);
+      } else {
+        // NOTE: here we are using _legacydata to preserve the original kind loading
+        // order if there is no sort function, which is totally weird.
```
Comment (tmeasday) [inline lib/client-api/src/story_store.ts]: I feel like this comment isn't really explaining why we had to do it..

**Review by tmeasday — APPROVED** [review]
LGTM, comment could maybe be improved.

**Comment by zenoplex** [comment]
https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#storysort-option

**Comment by benmvp** [comment]
@shilman I think this _may_ have caused the bug I've mentioned in #9477. I left some details there...

**Pull request merged**